pkg/constraint/constraint.go

```diff
@@ -17,11 +17,16 @@ limitations under the License.
 package constraint
 
 import (
+	"fmt"
+
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 )
 
 // Allowed returns true if pods is a collection of bound pods
 // which can run without conflict on a single minion.
-func Allowed(pods []api.BoundPod) bool {
-	return !PortsConflict(pods)
+func Allowed(pods []api.BoundPod) error {
+	if PortsConflict(pods) {
```

Comment:
While you are cleaning this up, can you rename PortsConflict to hostPortsConflict and the error message to "host port is already in use" - or even better, return a list of ports that are in conflict and include that in the error message!